source/features/check-if-follower.js

```diff
@@ -0,0 +1,28 @@
+import {h} from 'dom-chef';
+import select from 'select-dom';
+import {getUsername} from '../libs/utils';
+import api from '../libs/api';
+import {getCleanPathname} from '../libs/page-detect';
+
+export default async function () {
+	const username = getUsername();
+	const currentProfile = getCleanPathname();
+	const badge = <div class="follower-badge">Follows you</div>;
+
+	if (currentProfile === username) {
+		return;
+	}
+
+	const follower = await api(`users/${currentProfile}/following/${username}`).catch(error => {
```

Comment:
Use try/catch instead of `.catch`

Reply:
so something like

Reply:
so something like 

```
try {
		await api(`users/${currentProfile}/following/${username}`)
	} catch(err) {
		if(err.message.includes('end of JSON')) {
			select('.p-nickname').append(badge);
		}
	}

Reply:
> so something like?

Yes.

> I'm not sure how to squash the 404 errors here though.

You're already doing it in the above code block.

Reply:
preventing these from displaying
![screenshot from 2018-08-22 01-02-59](https://user-images.githubusercontent.com/3723666/44443817-d109c500-a5a7-11e8-88cf-01d0af110d6a.png)

Reply:
try/catch and `.catch()` should have the same behavior though

Reply:
My mistake, I thought the error from api.js:32 was being caught in @ 453c13c but it was due to me accidentally removing the await. Currently the snippet from the my 1st comment is showing both errors.

Reply:
So it makes sense api.js:32 doesn't get caught right? Since it's a console.log() and not a thrown error, but why is the one from line 18 not?